src/transformers/models/clap/modeling_clap.py

```diff
@@ -0,0 +1,2448 @@
+    models.
+    """
+
+    config_class = ClapTextConfig
+    base_model_prefix = "clap"
+    supports_gradient_checkpointing = False
+    _keys_to_ignore_on_load_missing = [r"position_ids", r"logit_scale_a", r"logit_scale_t"]
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        factor = self.config.initializer_factor
+
+        if isinstance(module, ClapTextEmbeddings):
+            module.word_embeddings.weight.data.normal_(mean=0.0, std=factor * 0.02)
+            module.position_embeddings.weight.data.normal_(mean=0.0, std=factor * 0.02)
+            module.token_type_embeddings.weight.data.normal_(mean=0.0, std=factor * 0.02)
+        elif isinstance(module, ClapTextSelfAttention):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.query.weight, std=in_proj_std)
+            nn.init.normal_(module.key.weight, std=in_proj_std)
+            nn.init.normal_(module.value.weight, std=in_proj_std)
+        elif isinstance(
+            module,
+            (
+                ClapTextSelfOutput,
+                ClapTextOutput,
+                ClapTextIntermediate,
+                ClapTextPooler,
+                ClapAudioSelfOutput,
+                ClapAudioIntermediate,
+                ClapAudioOutput,
+            ),
+        ):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.dense.weight, std=in_proj_std)
+        elif isinstance(module, ClapProjectionLayer):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.linear1.weight, std=in_proj_std)
+            nn.init.normal_(module.linear2.weight, std=in_proj_std)
+        elif isinstance(module, ClapAudioPatchEmbed):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.proj.weight, std=in_proj_std)
+        elif isinstance(module, ClapAudioSelfAttention):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.query.weight, std=in_proj_std)
+            nn.init.normal_(module.key.weight, std=in_proj_std)
+            nn.init.normal_(module.value.weight, std=in_proj_std)
+        elif isinstance(module, ClapAudioPatchMerging):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.reduction.weight, std=in_proj_std)
+        elif isinstance(module, ClapAudioEncoder):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.head.weight, std=in_proj_std)
```

Comment:
Looks like a lot of the linear layers are instantiated this way. Consider making it the default?

Reply:
yeah makes sense 💯

Reply:
Fixed in [230b516](https://github.com/huggingface/transformers/pull/21370/commits/230b5167942da06526c479ffdfd28b4892759bfb)